src/Symfony/Component/Serializer/Mapping/Loader/YamlFileLoader.php

```diff
@@ -87,11 +87,27 @@ public function loadClassMetadata(ClassMetadataInterface $classMetadata)
                 }
 
                 if (isset($data['serialized_name'])) {
-                    if (!\is_string($data['serialized_name']) || empty($data['serialized_name'])) {
-                        throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
+                    if (!\is_string($data['serialized_name']) && !\is_array($data['serialized_name'])) {
+                        throw new MappingException(sprintf('The "serialized_name" value must be a non-empty string or an array of serialized name/groups in "%s" for the attribute "%s" of the class "%s".', $this->file, $attribute, $classMetadata->getName()));
```

Comment:
Shouldn't a new key be introduced for the new shape? `serialized_names`?

Reply:
What do you mean ? We still talk about serialized_name definition in YAML

```php

// YAML 
  attributes:
    foo:
      serialized_name:
         bar: ~
         baz: ['a', 'b']
         bat: ['c']

Reply:
I mean `serialized_names` (with a final `s`) would be more coherent with what we have elsewhere in the component (for instance it's `groups` and not `group`).

Reply:
In YAML, we should use `serialized_names` (with an s). In XML, that would be multiple `<serialize-name>` entries (without an s) to be consistent with the way configuration works in Symfony indeed.